rxjava/src/main/kotlin/com/mindera/skeletoid/rxjava/Observable.kt

```diff
@@ -46,21 +53,40 @@ internal data class DataHolder<T>(val something: T? = null, val throwable: Throw
  * @param timeToWait - the time to wait
  * @param timeUnit - the unit of timeToWait
  */
-fun <T> Observable<T>.delayAtLeast(timeToWait: Long = 1000, timeUnit: TimeUnit = TimeUnit.MILLISECONDS): Observable<T> {
+fun <T> Observable<T>.delayAtLeast(
+    timeToWait: Long = 1000,
+    timeUnit: TimeUnit = TimeUnit.MILLISECONDS
+): Observable<T> {
+    var data: DataHolder<T>
     return Observable.zip<DataHolder<T>, Long, DataHolder<T>>(this.map {
-        DataHolder(
-            something = it
-        )
-    }.onErrorReturn { DataHolder(throwable = it) },
-            Observable.timer(timeToWait, timeUnit), BiFunction { t, _ -> t }).map { it.something ?: it.throwable?.let { throwable -> throw throwable } ?: throw Exception() }
+        data = DataHolder(something = it)
+        data
+    }
+        .onErrorReturn { DataHolder(throwable = it) },
+        Observable.timer(timeToWait, timeUnit), BiFunction { t, _ -> t })
+        .map {
+            it.something ?: throw it.throwable ?: Exception()
+        }
 }
 
 fun <T> Observable<T>.allowMultipleSubscribers(): Observable<T> =
-        share()
-                .replay(1)
-                .autoConnect(1)
-
-
+    share()
+        .replay(1)
+        .autoConnect(1)
+
+fun <T> Observable<T>.filterAndDo(condition: Boolean, runIfFiltered: () -> Unit): Observable<T> =
+    doOnNext {
+        if (condition) {
+            runIfFiltered()
+        }
+    }.filter { condition }
```

Comment:
I am very confused at this point, but if you want to have this code structure instead of my suggestion you should do two things:
- change the name to something like "filterOrDo"
- change the if to something like "if(!condition)"
You may be thinking that .filter(...) will filter when you return true but it's the other way around (check my code and its output in the previous comment about this)

Reply:
change the if to something like "if(!condition)" -> It's done, check the latest version of the code